osf/models/collection.py

```diff
@@ -39,6 +45,68 @@ class Meta:
     program_area = models.CharField(blank=True, max_length=127)
     school_type = models.CharField(blank=True, max_length=127)
     study_design = models.CharField(blank=True, max_length=127)
+    machine_state = models.IntegerField(
+        choices=CollectionSubmissionStates.int_field_choices(),
+        default=CollectionSubmissionStates.IN_PROGRESS,
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.state_machine = CollectionSubmissionMachine(
+            model=self,
+            active_state=self.state,
+            state_property_name='state'
+        )
+
+    @property
+    def state(self):
+        '''Property to translate between ApprovalState Enum and DB string.'''
+        return CollectionSubmissionStates(self.machine_state)
+
+    @property
+    def is_moderated(self):
+        return self.provider and self.provider.reviews_workflow == 'post-moderation'
+
+    @state.setter
+    def state(self, new_state):
+        self.machine_state = new_state.value
+
+    def _on_submit(self, event_data):
+        self.submitted_timestamp = timezone.now()
+
+    def _on_accept(self, event_data):
+        user = event_data.kwargs['user']
+        if user.has_perm('accept_submissions', self.provider):
+            raise PermissionsError(f'{user} must have moderator permissions.')
+
+    def _on_reject(self, event_data):
+        user = event_data.kwargs['user']
+        if user.has_perm('reject_submissions', self.provider):
+            raise PermissionsError(f'{user} must have moderator permissions.')
+
+    def _on_remove(self, event_data):
+        pass
+
+    def _on_resubmit(self, event_data):
+        pass
```

Comment:
Nit: Add a note that these are placeholders in case extra logic is needed?

Reply:
Yes it's used in the list endpoint PR, where the tests are.

Reply:
The callbacks don't do anything, though. You could technically remove them from the transition definitions and nothing would change